blocks/editable/index.js

```diff
@@ -474,12 +493,14 @@ export default class Editable extends Component {
 
 	onNodeChange( { parents } ) {
 		const formats = {};
-		const link = find( parents, ( node ) => node.nodeName.toLowerCase() === 'a' );
-		if ( link ) {
-			formats.link = { value: link.getAttribute( 'href' ) || '', target: link.getAttribute( 'target' ) || '', node: link };
-		}
-		const activeFormats = this.editor.formatter.matchAll( [	'bold', 'italic', 'strikethrough' ] );
-		activeFormats.forEach( ( activeFormat ) => formats[ activeFormat ] = true );
+		const formatNames = this.props.formattingControls || FORMATS;
+
+		forEach( this.editor.formatter.matchAll( formatNames ), ( activeFormat ) => {
```

Comment:
Should this be a `map` instead of `forEach`?

Reply:
reduce!